content/academy/puppeteer_playwright/collecting_data.md

```diff
@@ -48,7 +48,7 @@ const page = await browser.newPage();
 
 await page.goto('https://demo-webstore.apify.org/search/on-sale');
 
-await page.waitForNetworkIdle();
+await page.waitForNavigation({ waitUntil: 'DOMContentLoaded' });
```

Comment:
This isn't correct for multiple reasons.

- You don't need to `waitForNavigation` after `page.goto`. Especially not for `domcontentloaded`, because `goto` waits for `load` event by default, which fires after `domcontentloaded`.
   - You only need to wait for navigation BEFORE you click something which triggers a navigation and then await the promise after the click. It's all in the documentation examples I sent you.
- You can put the `waitUntil` directly in the `goto` function. No need for extra function.
- It's `domcontentloaded` not `DOMContentLoaded`. Maybe it works as well, but in the docs its lowercased.

Reply:
Also, if we're changing the default load eventg from `load` to `domcontentloaded` we should explain why we're doing that.

Reply:
The fact that `load` is already waited for after `goto` is fired cleared this up